pennylane/pulse/convenience_functions.py

```diff
@@ -134,37 +163,69 @@ def f(p, t):
               If an integer is provided, the timespan is defined as ``(0, timespan)``.
 
     Returns:
             func: a function that takes two arguments, an array of trainable parameters and a `float` defining the
-            time at which the function is evaluated. When called, the function uses the array of parameters to
-            create evenly sized bins within the ``timespan``, with each bin value set by an element of the array.
-            It then selects the value of the parameter array corresponding to the specified time, based on the
-            assigned binning.
+            time at which the function is evaluated.
+
+    This function can be used to create a parametrized coefficient function that is piecewise constant
+    within the interval ``t``, and 0 outside it. When creating the callable, only the timespan is passed. The number
+    of bins and values for the parameters are set when ``params`` is passed to the callable. Each bin value is set by
+    an element of the ``params`` array. The variable ``t`` is used to select the value of the parameter array
+    corresponding to the specified time, based on the assigned binning.
+
+    .. figure:: ../../_static/pulse/pwc_example.png
+        :align: center
+        :width: 60%
+        :target: javascript:void(0);
 
     **Example**
 
-    >>> timespan = (1, 3)
-    >>> f1 = pwc(timespan)
+    >>> timespan = (2, 7)
+    >>> f1 = qml.pulse.pwc(timespan)
 
     The resulting function ``f1`` has the call signature ``f1(params, t)``. If passed an array of parameters and
     a time, it will assign the array as the constants in the piecewise function, and select the constant corresponding
     to the specified time, based on the time interval defined by ``timespan``.
 
-    >>> params = [10, 11, 12, 13, 14]
-    >>> f1(params, 2)
-    Array(12, dtype=int32)
+    .. code-block:: python
+
+        >>> H = f1 * qml.PauliX(0)
+
+        # passing pwc((2, 7)) an array evenly distributes the array values in the interval t=2 to t=7
+        >>> H(params=[[11, 12, 13, 14, 15]], t=2.3)
+        11.0*(PauliX(wires=[0]))
+
+        # different time, same bin, same result
+        >>> H(params=[[1, 2, 3, 4, 5]], t=2.5)
+        11.0*(PauliX(wires=[0]))
+
+        # next bin
+        >>> H(params=[[1, 2, 3, 4, 5]], t=3.1)
+        12.0*(PauliX(wires=[0]))
 
-    >>> f1(params, 2.1)  # same bin
-    Array(12, dtype=int32)
+        # outside the window - the function is assigned non-zero values
+        >>> H(params=[[1, 2, 3, 4, 5]], t=8)
+        0.0*(PauliX(wires=[0]))
+
+    .. note::
+        The final time in the timespan indicates the index at which the function output switches from params[-1] to 0.
+        As such, the final time in ``timespan`` returns 0:
+
+        >>> H(params=[[1, 2, 3, 4, 5]], t=6.999999)
+        15.0*(PauliX(wires=[0]))
+
+        >>> H(params=[[1, 2, 3, 4, 5]], t=7)
+        0.0*(PauliX(wires=[0]))
```

Comment:
I think we should make this a warning and put it higher up

Reply:
Moved to just above `**Example**` and made it a warning